File: app/src/main/java/com/veyndan/paper/reddit/ui/widget/PostFlairsLayout.java
package com.veyndan.paper.reddit.ui.widget;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;

import com.google.android.flexbox.FlexboxLayout;
import com.veyndan.paper.reddit.databinding.PostFlairBinding;
import com.veyndan.paper.reddit.post.Flair;

import java.util.Collection;

public class PostFlairsLayout extends FlexboxLayout {

    public PostFlairsLayout(final Context context, final AttributeSet attrs) {
        super(context, attrs);
    }

    public void setFlairs(final Collection<Flair> flairs, final String subreddit) {
        if (flairs.isEmpty()) {
            setVisibility(GONE);
            return;
        }

        for (final Flair flair : flairs) {
            final PostFlairBinding binding = PostFlairBinding.inflate(LayoutInflater.from(getContext()), this, false);
            binding.postFlair.setFlair(flair, subreddit);
            addView(binding.getRoot());
        }

        setVisibility(VISIBLE);
    }
}

File: app/src/main/java/com/veyndan/paper/reddit/ui/widget/PostFlairsLayout.kt
package com.veyndan.paper.reddit.ui.widget

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View

import com.google.android.flexbox.FlexboxLayout
import com.veyndan.paper.reddit.databinding.PostFlairBinding
import com.veyndan.paper.reddit.post.Flair

class PostFlairsLayout(context: Context, attrs: AttributeSet) : FlexboxLayout(context, attrs) {

    fun setFlairs(flairs: Collection<Flair>, subreddit: String) {
        removeAllViews()

        if (flairs.isEmpty()) {
            visibility = View.GONE
            return
        }

        for (flair in flairs) {
            val binding = PostFlairBinding.inflate(LayoutInflater.from(context), this, false)
            binding.postFlair.setFlair(flair, subreddit)
            addView(binding.root)
        }

        visibility = View.VISIBLE
    }
}
